tests/test_serialization.py

```diff
@@ -0,0 +1,248 @@
+	M.set_species(x0)
+
+
+    # Finally, to test assignment rules, we'll add a rule adding an "inducer" at 
+    # time 75.
+	M.create_parameter("I0", 10) #Inducer concentration
+	M.create_parameter("T_I0", 2) #Initial time inducer is added
+	M.create_rule("assignment", {"equation":"I = _I0*Heaviside(t-_T_I0)"})
+
+
+	###############################
+	# TEST SERIALIZATION OF MODEL #
+	###############################
+	timepoints = np.arange(0, 25, 0.01)
+	prepickling_results = py_simulate_model(timepoints, 
+									Model = M, 
+									stochastic = True, 
+									return_dataframe = False).py_get_result()
+	pickled_M = pickle.dumps(M)
+	postpickled_M = pickle.loads(pickled_M)
+
+	test_utils.set_seed(seed)
+	postpickling_results = py_simulate_model(timepoints, 
+									Model = postpickled_M, 
+									stochastic = True, 
+									return_dataframe = False).py_get_result()
+
+	assert np.allclose(prepickling_results, postpickling_results), \
+		   	"Complex Model changes when pickled and unpickled."
+
+
+def test_random_complex_lineagemodel():
+	'''
+	Creates a two-reaction LineageModel with more or less the same settings as
+	the Model in test_random_complex_model, plus linage_propensities, a volume 
+	event, a division rule, and a death rule.
+
+	This test currently does NOT distinguish between pickling and depickling
+	errors -- it just checks that if you pickle and depickle a Model, it still
+	produces the same results when simulated.
+	'''
+
+	test_utils.set_seed(seed)
+
+	################
+	# CREATE MODEL #
+	################
+
+	# Start with a basic reaction A+B-->C
+	inputs = ["A", "B"]
+	outputs = ["C"]
+	all_species = inputs + outputs
+	param_min, param_max = -4, 4
+
+	if debug:
+		print('simulating propensity type ', prop_type)
+
+	
+	# We will use a random(ish) rational function for propensity
+	param_dict = {}
+	rate_str = "(1+"
+	numerator_terms = np.random.randint(0, 5)
+	denominator_terms = np.random.randint(0, 5)
+	for i in range(numerator_terms):
+		coef = str(round(np.exp(np.random.uniform(low = param_min,
+												 high = param_max)), 3))
+		exp = str(round(np.random.uniform(low = 0,high = param_max), 3))
+		species = all_species[np.random.randint(len(all_species))]
+		rate_str += coef + "*" + species + "^" + exp + "+"
+	rate_str = rate_str[:-1] + ")"
+	rate_str += "/(1+"
+	for i in range(denominator_terms):
+		coef =str(round(np.exp(np.random.uniform(low = param_min,
+												 high = param_max)), 3))
+		exp = str(round(np.random.uniform(low = 0,high = param_max), 3))
+		species = all_species[np.random.randint(len(all_species))]
+		rate_str += coef + "*" + species + "^" + exp + "+"
+	rate_str =  rate_str[:-1] + ")"
+	param_dict['rate'] = rate_str
+	if debug:
+		print('\t params =', param_dict)
+
+
+	### DELETEME
+	param_dict["rate"] = "A*3"
+
+	consolidation_rxn = (inputs, outputs, "general", param_dict)
+
+	# We'll add an A-->B reaction with delay.
+	delay_k = round(np.exp(np.random.uniform(low = -1,
```

Comment:
Delays are actually not currently supported in lineages due to complications partitioning the delay queue objects, so don't need to include this in the tests.